### PR TITLE
Clean up dependencies

### PR DESCRIPTION
**Pull request by boswelja**
[comment]: # (Replace [ ] with [x].)
- [x] Read [`CONTRIBUTING.md`](https://github.com/imashnake0/Animite/blob/be53ba4561c8ff20f588fb348965f208e301b6b8/CONTRIBUTING.md).

Simplified Compose dependency declaration & removed unused deps

**Summary of changes:**

1. Removed chrisbanes/snapper
2. Removed kotlin-reflect
3. ~~Explicitly defined~~ Removed compose-material-icons
4. Created a bundle for Compose and removed the BOM

**Tested changes:**

App still builds and runs

[comment]: # (THANK YOU! ♡\(灬♥ ◡ ♥灬\))

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -114,5 +112,15 @@ hilt-android-compiler = { group = "com.google.dagger", name = "hilt-android-comp
...
 accompanist-placeholder = { group = "com.google.accompanist", name = "accompanist-placeholder", version.ref = "accompanist" }
-chrisbanes-snapper = { group = "dev.chrisbanes.snapper", name = "snapper", version.ref = "snapper" }
 test-junit = { group = "junit", name = "junit", version.ref = "junit" }
+
+[bundles]
+
+compose = [
+    "compose-animation",
+    "compose-foundation",
+    "compose-material-icons",
```
Comment (imashnake0) [inline gradle/libs.versions.toml:122]: Should we remove this from the bundle since it's not necessarily used if compose is?

Comment (boswelja) [reply gradle/libs.versions.toml:122]: We can, but is it not necessarily used because we haven't implemented most of the stuff?

Comment (imashnake0) [reply gradle/libs.versions.toml:122]: https://developer.android.com/reference/kotlin/androidx/compose/material/icons/package-summary:
> The most commonly used set of Material icons are provided by androidx.compose.material:material-icons-core - this library is also provided as an API dependency by androidx.compose.material:material.

> A separate library, androidx.compose.material:material-icons-extended, contains the full set of Material icons. Due to the very large size of this library, make sure to use R8/Proguard to strip unused icons if you are including this library as a direct dependency. Alternatively you can make a local copy (by copy and pasting) the icon(s) you wish to keep, or using Android Studio's 'Import vector asset' feature.

I wonder where it's used :thinkies:

Comment (boswelja) [reply gradle/libs.versions.toml:122]: All I know is random icons aren't available without it lul

Comment (imashnake0) [reply gradle/libs.versions.toml]: Removed it.

**Review by imashnake0 — APPROVED** [review]
Thank you!

**Pull request merged**